### ISSUE TITLE
feature request: service to clean up old device ids

### ISSUE THREAD

**Issue by wuppiwuppi** [zachcheatham/ha-omada#49]
Hi,

I like that integration really much - however I have a lot of "tracked devices". 
The AVM FritzTools Integration offers a service to clean up old entities.  https://www.home-assistant.io/integrations/fritz/#platform-services

It would be nice to have a similar thing for this integration. 

br 
Andre

**Comment by jcbolive** [comment]
I would also like a function to filter the devices for integration

**Comment by wuppiwuppi** [comment]
You can always disable the not needed ones in home assistant. 

How would you define the Filter?

**Comment by jcbolive** [comment]
I would like to filter like it is done below, inside the config

  Omada:
    filter:
      include_entities:
        - entidie_ID

**Comment by ankushg** [comment]
+1 to this - would be great if we could opt-in to only track certain devices

**Comment by TommySharpNZ** [comment]
Agreed, rather than blindly loading all devices.... I would prefer to select which devices the integration tracks.

**Comment by J1mooo** [comment]
That would be a cool feature +1

**Comment by martinj27** [comment]
+1

**Comment by nafur** [comment]
I'd love some version of this. My HA interface is virtually unusable once it has loaded all omada devices...
Some other variants that would solve this for me:
- automatically delete devices that omada no longer lists as clients (and are not used in any automation)
- automatically delete devices that we haven't seen for X days
- only add devices that omada lists as clients with an IP (don't ask...)
- only add devices with IPs in a configurable subnet

**Comment by nafur** [comment]
So I looked into this a bit further, and realized that there is a lot of stuff already in place to deal with this issue: there is an ssid filter (I'm not sure about the relation of ssid and vlan names here), and there are mechanisms to remove "outdated" entities.
~There is one weird thing so: The core logic of which devices are considered is here:~
https://github.com/zachcheatham/ha-omada/blob/257cccca035b1315b0ef2f0e7160b926c1b96a8b/custom_components/omada/controller.py#L184
~It says to add a device if there is no ssid filter set *or* the client was not already added *or* the ssid of the client is in the ssid filter. This seems a bit off, and I would expect this to be if there is no ssid filter set *or* (the client was not already added *and* the ssid of the client is in the ssid filter).
I've done this change locally, and it seems to reduce the number of (stale) devices and entities in my system every time HA is restarted (it will always timeout because there are so many, but it manages to remove a bunch before).
@zachcheatham is this a bug?~

[Update] After some more debugging, it seems that this second part of the condition takes care of devices (in the omada sense) that are not clients, i.e., switches, APs etc. So I guess as long as you can get away with the ssid filter, everything should be fine.
